xgboost_ray/data_sources/modin.py

```diff
@@ -0,0 +1,52 @@
+from typing import Any, Optional, Sequence
+
+from xgboost_ray.data_sources.data_source import DataSource, RayFileType
+
+import pandas as pd
+
+try:
+    import modin  # noqa: F401
+    MODIN_INSTALLED = True
+except ImportError:
+    MODIN_INSTALLED = False
+
+
+class Modin(DataSource):
+    @staticmethod
+    def is_data_type(data: Any,
+                     filetype: Optional[RayFileType] = None) -> bool:
+        if not MODIN_INSTALLED:
+            return False
+        from modin.pandas import DataFrame as ModinDataFrame, \
+            Series as ModinSeries
+
+        return isinstance(data, (ModinDataFrame, ModinSeries))
+
+    @staticmethod
+    def load_data(
```

Comment:
do you need to check modin installation for all functions?

Reply:
No - if the data source is incompatible it will not be selected when trying to load data:

```
            if source.is_data_type(self.data, self.filetype):
                data_source = source
                break
```

Reply:
But I guess it can't hurt to add an `_assert_modin_installed()`